python/cudf/cudf/utils/dtypes.py

```diff
@@ -494,6 +494,28 @@ def find_common_type(dtypes):
     if len(dtypes) == 0:
         return None
 
+    # Early exit for categoricals since they're not hashable and therefore
+    # can't be put in a set.
+    if any(cudf.api.types.is_categorical_dtype(dtype) for dtype in dtypes):
```

Comment:
I think the below only makes sense when all are unordered categoricals.

Reply:
That's a great point, I'll add that check.